src/c_api/object_detection_2d_nanodet_jit.cpp

```diff
@@ -0,0 +1,357 @@
+ */
+torch::DeviceType torchDevice(char *deviceName, int verbose = 0) {
+  torch::DeviceType device;
+  if (std::string(deviceName) == "cuda") {
+    if (verbose == 1)
+      printf("to cuda\n");
+    device = torch::kCUDA;
+  } else {
+    if (verbose == 1)
+      printf("to cpu\n");
+    device = torch::kCPU;
+  }
+  return device;
+}
+
+void loadNanodetModel(char *modelPath, char *device, int height, int width, float scoreThreshold, NanodetModelT *model) {
+  // Initialize model
+  model->inputSizes[0] = width;
+  model->inputSizes[1] = height;
+
+  model->scoreThreshold = scoreThreshold;
+  model->keepRatio = 1;
+
+  const std::vector<std::string> labels{
+    "person",         "bicycle",    "car",           "motorcycle",    "airplane",     "bus",           "train",
+    "truck",          "boat",       "traffic light", "fire hydrant",  "stop sign",    "parking meter", "bench",
+    "bird",           "cat",        "dog",           "horse",         "sheep",        "cow",           "elephant",
+    "bear",           "zebra",      "giraffe",       "backpack",      "umbrella",     "handbag",       "tie",
+    "suitcase",       "frisbee",    "skis",          "snowboard",     "sports ball",  "kite",          "baseball bat",
+    "baseball glove", "skateboard", "surfboard",     "tennis racket", "bottle",       "wine glass",    "cup",
+    "fork",           "knife",      "spoon",         "bowl",          "banana",       "apple",         "sandwich",
+    "orange",         "broccoli",   "carrot",        "hot dog",       "pizza",        "donut",         "cake",
+    "chair",          "couch",      "potted plant",  "bed",           "dining table", "toilet",        "tv",
+    "laptop",         "mouse",      "remote",        "keyboard",      "cell phone",   "microwave",     "oven",
+    "toaster",        "sink",       "refrigerator",  "book",          "clock",        "vase",          "scissors",
+    "teddy bear",     "hair drier", "toothbrush"};
+
+  // mean and standard deviation tensors for normalization of input
+  torch::Tensor meanTensor = torch::tensor({{{-103.53f}}, {{-116.28f}}, {{-123.675f}}});
+  torch::Tensor stdValues = torch::tensor({{{0.017429f}}, {{0.017507f}}, {{0.017125f}}});
+
+  // initialization of jit model and class as holder of c++ values.
+  torch::DeviceType initDevice = torchDevice(device, 1);
+  torch::jit::script::Module network = torch::jit::load(modelPath, initDevice);
+  network.eval();
+
+  NanoDet *detector = new NanoDet(network, meanTensor, stdValues, initDevice, labels);
+
+  model->network = static_cast<void *>(detector);
+}
+
+void ffNanodet(NanoDet *model, torch::Tensor *inputTensor, cv::Mat *warpMatrix, cv::Size *originalSize,
+               torch::Tensor *outputs) {
+  // Make all the inputs as tensors to use in jit model
+  torch::Tensor srcHeight = torch::tensor(originalSize->width);
+  torch::Tensor srcWidth = torch::tensor(originalSize->height);
```

Comment:
width is being assigned to height and height to width. Is this intentional?

Reply:
The code has been modified to align with the convention of having the width and height in a different order, as is common in Python. However, I will revert the change in order to maintain clarity and consistency with the C convention.

Reply:
Alright, i just commented this in case it was done by mistake.